pwnlib/tubes/ssh.py

```diff
@@ -1560,6 +1565,7 @@ def download_dir(self, remote=None, local=None):
             local_tar = tempfile.NamedTemporaryFile(suffix='.tar.gz')
             self.download_file(remote_tar, local_tar.name)
 
+            self.system(b"rm "+remote_tar)
```

Comment:
We should try SFTP first, then `sh_string(remote_tar)` like above. See `download_file()`.

Reply:
Does this new commit address the issue?